tests/util/base.py

```diff
@@ -0,0 +1,117 @@
+"""
+Base test classes for PyLTC integration testing.
+
+"""
+
+from pyltc import DIR_EGRESS
+from pyltc.parseargs import parse_args
+from pyltc.target import TcFileTarget, TcCommandTarget
+from pyltc.main import pyltc_entry_point
+from tests.util.iperf_proc import TCPNetPerfTest, UDPNetPerfTest
+from pyltc.plugins.util import parse_branch
+
+
+class TcTestTarget(TcFileTarget):
+    """A Target class that sends generated commands to caller
+    instead of executing/writing in file. Used for testing purposes."""
+
+    def __init__(self, iface, callback, direction=DIR_EGRESS):
+        TcFileTarget.__init__(self, iface, direction=direction)
+        self._callback = callback
+
+    def install(self, verbose=False):
+        self._callback(self._commands)
+
+
+class LtcSimulateTargetRun(object):
+    """Used by simulation pyltc tests.
+    Executes given pyltc command using the TcTestTarget."""
+
+    def __init__(self, argv, full=False):
+        self._argv = argv
+        self._full = full
+        self._result = None
+
+    def our_callback(self, result):
+        self._result = result
+
+    def test_target_factory(self, iface, direction):
+        return TcTestTarget(iface, self.our_callback, direction)
+
+    def run(self):
+        pyltc_entry_point(self._argv, self.test_target_factory)
+
+    @property
+    def result(self):
+        return self._result
+
+
+class LtcLiveTargetRun(object):
+    """Used by live pyltc tests.
+    Executes given pyltc command using TcCommandTarget
+    and measures rates, then returns results."""
+
+    def __init__(self, argv, udp_sendrate, duration=5, full=False):
+        self._argv = argv
+        self._udp_sendrate = udp_sendrate
+        self._duration = duration
+        self._full = full
+        self._result = None
+
+    def _target_factory(self, iface, direction):
+        return TcCommandTarget(iface, direction)
+
+    def _test_for_portrange(self, klass, port_range, sendrate):
+        if '-' in port_range:
+            left_in_port, right_in_port = map(int, port_range.split('-'))
+            left_out_port = left_in_port - 1
+            right_out_port = right_in_port + 1
```

Comment:
This may also be more elegantly implemented using regex. The current implementation is also fine, I just think regex would be cleaner.